src/components/timeline/TrackPowerGraph.js

```diff
@@ -129,34 +130,74 @@ class TrackPowerCanvas extends React.PureComponent<CanvasProps> {
       ctx.fillStyle = '#73737388'; // Grey 50 with transparency.
       ctx.beginPath();
 
-      // The x and y are used after the loop.
-      let x = 0;
-      let y = 0;
-      let firstX = 0;
-      for (let i = sampleStart; i < sampleEnd; i++) {
-        // Create a path for the top of the chart. This is the line that will have
-        // a stroke applied to it.
-        x = (samples.time[i] - rangeStart) * millisecondWidth;
+      const getX = (i) =>
+        Math.round((samples.time[i] - rangeStart) * millisecondWidth);
```

Comment:
So differently, we are rounding the numbers with this patch. I guess this is going to change the shape of the graph a bit right? But otherwise it's pretty much impossible to implement this without rounding, and it looks like the shape doesn't change that much, so looks good to me.

Reply:
Even when switching back and forth between the same profile on profiler.firefox.com and on my local version with the patch applied, I couldn't see any difference in the shape of the graph. The deviceWidth is `width * devicePixelRatio`, so any integer x value here is already a half pixel on a hidpi screen. I think that's the reason why there's no visible difference.